### PR TITLE
Feat: Add default permissions to most plugins

### PR DESCRIPTION
**Pull request by tillmann-crabnebula**
This PR adds default permissions to most plugins which have functionality exposed to the frontend.

As of now mobile plugins are a special case and not all of them have default permissions and use the command macro. These need to be investigated before adding default permissions.

One thing I noticed is that these sections of the documentation are not auto generated from the permission description and we need to add some automation similar to config reference docs.

### CODE REVIEW

**Comment by naman-crabnebula** [comment]
Hi!

This is not working. It is showing error

```
  Permission notification:allow-register-action-types not found, expected one of app:default, app:allow-app-hide, app:allow-app-show, app:allow-name, app:allow-tauri-version, app:allow-version, app:deny-app-hide, app:deny-app-show, app:deny-name, app:deny-tauri-version, app:deny-version, event:default, event:allow-emit, event:allow-emit-to, event:allow-listen, event:allow-unlisten, event:deny-emit, event:deny-emit-to, event:deny-listen, event:deny-unlisten, image:default, image:allow-from-bytes, image:allow-from-path, image:allow-new, image:allow-rgba, image:allow-size, image:deny-from-bytes, image:deny-from-path, image:deny-new, image:deny-rgba, image:deny-size, menu:default, menu:allow-append, menu:allow-create-default, menu:allow-get, menu:allow-insert, menu:allow-is-checked, menu:allow-is-enabled, menu:allow-items, menu:allow-new, menu:allow-popup, menu:allow-prepend, menu:allow-remove, menu:allow-remove-at, menu:allow-set-accelerator, menu:allow-set-as-app-menu, menu:allow-set-as-help-menu-for-nsapp, menu:allow-set-as-window-menu, menu:allow-set-as-windows-menu-for-nsapp, menu:allow-set-checked, menu:allow-set-enabled, menu:allow-set-icon, menu:allow-set-text, menu:allow-text, menu:deny-append, menu:deny-create-default, menu:deny-get, menu:deny-insert, menu:deny-is-checked, menu:deny-is-enabled, menu:deny-items, menu:deny-new, menu:deny-popup, menu:deny-prepend, menu:deny-remove, menu:deny-remove-at, menu:deny-set-accelerator, menu:deny-set-as-app-menu, menu:deny-set-as-help-menu-for-nsapp, menu:deny-set-as-window-menu, menu:deny-set-as-windows-menu-for-nsapp, menu:deny-set-checked, menu:deny-set-enabled, menu:deny-set-icon, menu:deny-set-text, menu:deny-text, notification:default, notification:allow-is-permission-granted, notification:allow-notify, notification:allow-request-permission, notification:deny-is-permission-granted, notification:deny-notify, notification:deny-request-permission, path:default, path:allow-basename, path:allow-dirname, path:allow-extname, path:allow-is-absolute, path:allow-join, path:allow-normalize, path:allow-resolve, path:allow-resolve-directory, path:deny-basename, path:deny-dirname, path:deny-extname, path:deny-is-absolute, path:deny-join, path:deny-normalize, path:deny-resolve, path:deny-resolve-directory, resources:default, resources:allow-close, resources:deny-close, shell:allow-execute, shell:allow-kill, shell:allow-open, shell:allow-spawn, shell:allow-stdin-write, shell:deny-execute, shell:deny-kill, shell:deny-open, shell:deny-spawn, shell:deny-stdin-write, tray:default, tray:allow-get-by-id, tray:allow-new, tray:allow-remove-by-id, tray:allow-set-icon, tray:allow-set-icon-as-template, tray:allow-set-menu, tray:allow-set-show-menu-on-left-click, tray:allow-set-temp-dir-path, tray:allow-set-title, tray:allow-set-tooltip, tray:allow-set-visible, tray:deny-get-by-id, tray:deny-new, tray:deny-remove-by-id, tray:deny-set-icon, tray:deny-set-icon-as-template, tray:deny-set-menu, tray:deny-set-show-menu-on-left-click, tray:deny-set-temp-dir-path, tray:deny-set-title, tray:deny-set-tooltip, tray:deny-set-visible, webview:default, webview:allow-create-webview, webview:allow-create-webview-window, webview:allow-internal-toggle-devtools, webview:allow-print, webview:allow-reparent, webview:allow-set-webview-focus, webview:allow-set-webview-position, webview:allow-set-webview-size, webview:allow-set-webview-zoom, webview:allow-webview-close, webview:allow-webview-position, webview:allow-webview-size, webview:deny-create-webview, webview:deny-create-webview-window, webview:deny-internal-toggle-devtools, webview:deny-print, webview:deny-reparent, webview:deny-set-webview-focus, webview:deny-set-webview-position, webview:deny-set-webview-size, webview:deny-set-webview-zoom, webview:deny-webview-close, webview:deny-webview-position, webview:deny-webview-size, window:default, window:allow-available-monitors, window:allow-center, window:allow-close, window:allow-create, window:allow-current-monitor, window:allow-cursor-position, window:allow-destroy, window:allow-hide, window:allow-inner-position, window:allow-inner-size, window:allow-internal-toggle-maximize, window:allow-is-closable, window:allow-is-decorated, window:allow-is-focused, window:allow-is-fullscreen, window:allow-is-maximizable, window:allow-is-maximized, window:allow-is-minimizable, window:allow-is-minimized, window:allow-is-resizable, window:allow-is-visible, window:allow-maximize, window:allow-minimize, window:allow-monitor-from-point, window:allow-outer-position, window:allow-outer-size, window:allow-primary-monitor, window:allow-request-user-attention, window:allow-scale-factor, window:allow-set-always-on-bottom, window:allow-set-always-on-top, window:allow-set-closable, window:allow-set-content-protected, window:allow-set-cursor-grab, window:allow-set-cursor-icon, window:allow-set-cursor-position, window:allow-set-cursor-visible, window:allow-set-decorations, window:allow-set-effects, window:allow-set-focus, window:allow-set-fullscreen, window:allow-set-icon, window:allow-set-ignore-cursor-events, window:allow-set-max-size, window:allow-set-maximizable, window:allow-set-min-size, window:allow-set-minimizable, window:allow-set-position, window:allow-set-progress-bar, window:allow-set-resizable, window:allow-set-shadow, window:allow-set-size, window:allow-set-skip-taskbar, window:allow-set-title, window:allow-set-visible-on-all-workspaces, window:allow-show, window:allow-start-dragging, window:allow-start-resize-dragging, window:allow-theme, window:allow-title, window:allow-toggle-maximize, window:allow-unmaximize, window:allow-unminimize, window:deny-available-monitors, window:deny-center, window:deny-close, window:deny-create, window:deny-current-monitor, window:deny-cursor-position, window:deny-destroy, window:deny-hide, window:deny-inner-position, window:deny-inner-size, window:deny-internal-toggle-maximize, window:deny-is-closable, window:deny-is-decorated, window:deny-is-focused, window:deny-is-fullscreen, window:deny-is-maximizable, window:deny-is-maximized, window:deny-is-minimizable, window:deny-is-minimized, window:deny-is-resizable, window:deny-is-visible, window:deny-maximize, window:deny-minimize, window:deny-monitor-from-point, window:deny-outer-position, window:deny-outer-size, window:deny-primary-monitor, window:deny-request-user-attention, window:deny-scale-factor, window:deny-set-always-on-bottom, window:deny-set-always-on-top, window:deny-set-closable, window:deny-set-content-protected, window:deny-set-cursor-grab, window:deny-set-cursor-icon, window:deny-set-cursor-position, window:deny-set-cursor-visible, window:deny-set-decorations, window:deny-set-effects, window:deny-set-focus, window:deny-set-fullscreen, window:deny-set-icon, window:deny-set-ignore-cursor-events, window:deny-set-max-size, window:deny-set-maximizable, window:deny-set-min-size, window:deny-set-minimizable, window:deny-set-position, window:deny-set-progress-bar, window:deny-set-resizable, window:deny-set-shadow, window:deny-set-size, window:deny-set-skip-taskbar, window:deny-set-title, window:deny-set-visible-on-all-workspaces, window:deny-show, window:deny-start-dragging, window:deny-start-resize-dragging, window:deny-theme, window:deny-title, window:deny-toggle-maximize, window:deny-unmaximize, window:deny-unminimize
```

It is not recommended to manually edit the files, and just add them to `build.rs` of each plugin and run `pnpm build` to automatically generate the files.

**Comment by FabianLars** [comment]
These 2 lines were added with your PR in mind. With the correct merge order there won't be any issues.

**Comment by FabianLars** [comment]
Also, as far as I know build.rs can't generate the default.toml file. That's always up to the author, which is probably why there weren't any default permissions yet.

**Comment by tillmann-crabnebula** [comment]
We can generate the default permission as well during build time, but it is easier to contribute and fix when we add them manually.
For people not involved in the plugin development this is easier to understand than reading the `build.rs` generation code.

I think default permissions are the one case where it is okay to have manual work involved as these are considered "secure by default" and automating this and auto-including new commands will make it simpler to weaken the default.

File: plugins/autostart/permissions/default.toml
Comment (amrbashir) [inline plugins/autostart/permissions/default.toml:13]: this multiline description is messing up the generated table, see https://github.com/tillmann-crabnebula/plugins-workspace/blob/feat/default-permissions/plugins/autostart/permissions/autogenerated/reference.md, not sure if we can fix that or maybe keep the description to a single line?

Comment (tillmann-crabnebula) [reply plugins/autostart/permissions/default.toml:13]: I actually have some code locally that fixes this by wrapping the markdown into an html table but would address this in another PR.

Comment (tillmann-crabnebula) [reply plugins/autostart/permissions/default.toml:13]: For reference https://github.com/tauri-apps/tauri/pull/10113

Comment (FabianLars) [reply plugins/autostart/permissions/default.toml:13]: Should we wait for 10113 before we merge this one here then?

Comment (tillmann-crabnebula) [reply plugins/autostart/permissions/default.toml:13]: I think it does not matter too much as the generated markdown is not consumed anywhere yet.

**Review by amrbashir — APPROVED** [review]

**Pull request merged**
